jobs/build/ocp/Jenkinsfile

```diff
@@ -187,285 +242,346 @@ node(TARGET_NODE) {
+                    sh "ose_images.sh --user ocp-build update_docker --branch rhaos-${BUILD_VERSION}-rhel-7 --group base --force --release '${NEW_DOCKERFILE_RELEASE}' --version 'v${NEW_VERSION}'"
                 }
-            }
+                // End old method
 
-            stage( "puddle: ose 'building'" ) {
-              buildlib.build_puddle(
-                      PUDDLE_CONF,    // The puddle configuration file to use
-                      PUDDLE_SIGN_KEYS, // openshifthosted key
-                      "-b",   // do not fail if we are missing dependencies
-                      "-d",   // print debug information
-                      "-n",   // do not send an email for this puddle
-                      "-s",   // do not create a "latest" link since this puddle is for building images
-                      "--label=building"   // create a symlink named "building" for the puddle
-              )
-            }
+                // OIT Method
+                OIT_WORKING = "${pwd(tmp=true)}"
 
-            stage( "compare dist-git" ) {
-                sh "ose_images.sh --user ocp-build compare_nodocker --branch rhaos-${BUILD_VERSION}-rhel-7 --group base"
-            }
+                stage( "update dist-git" ) {
+                  sh "echo 'ose: ${env.OSE_DIR}' > ${env.WORKSPACE}/sources.yml"
+                  sh "echo 'jenkins: ${env.OPENSHIFT_JENKINS_DIR}' >> ${env.WORKSPACE}/sources.yml"
+                  sh "echo 'openshift-ansible: ${env.OPENSHIFT_ANSIBLE_DIR}' >> ${env.WORKSPACE}/sources.yml"
 
-            stage( "update dist-git" ) {
-                sh "ose_images.sh --user ocp-build update_docker --branch rhaos-${BUILD_VERSION}-rhel-7 --group base --force --release '${NEW_DOCKERFILE_RELEASE}' --version 'v${NEW_VERSION}'"
-            }
+                  oit """
+--working-dir ${OIT_WORKING} --group 'openshift-${BUILD_VERSION}' \\
+distgits:rebase --sources ${env.WORKSPACE}/sources.yml --version ${NEW_VERSION} \\
+--release ${NEW_DOCKERFILE_RELEASE} --no-push
+"""
+                }
 
-            stage( "build images" ) {
-                // TODO: Create a dynamic .repo file pointing to the exact puddle we built instead of "building" so that we can run X.Y builds in parallel
-                sh "ose_images.sh --user ocp-build build_container --branch rhaos-${BUILD_VERSION}-rhel-7 --group base --repo https://raw.githubusercontent.com/openshift/aos-cd-jobs/master/build-scripts/repo-conf/aos-unsigned-building.repo"
-            }
+                record_log = buildlib.parse_record_log( OIT_WORKING )
+                distgit_notify = buildlib.get_distgit_notify( record_log )
+
+                for(kv in buildlib.mapToList(distgit_notify)) {
+                    distgit = kv[0]
+                    val = kv[1]
+                    if(val.owners) {
+                        alias = val['source_alias']
+                        dockerfile_url = ""
+                        if ( alias ){
```

Comment:
When would we expect this to be missing?

Reply:
I thought some of the config.yml files had no source alias.

Reply:
If they don't have an alias, they can't be rebased and there will be no notify entry.